jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt

```diff
@@ -22,20 +29,51 @@ import software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MAX_TIMEOUT
 import software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MEMORY_INCREMENT
 import software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MIN_MEMORY
 import software.aws.toolkits.jetbrains.services.lambda.LambdaLimits.MIN_TIMEOUT
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamCommon
 import software.aws.toolkits.jetbrains.ui.SliderPanel
 import java.util.concurrent.TimeUnit
 
 object Lambda {
+    private val LOG = getLogger<Lambda>()
+
     fun findPsiElementsForHandler(project: Project, runtime: Runtime, handler: String): Array<NavigatablePsiElement> {
         val resolver = runtime.runtimeGroup?.let { LambdaHandlerResolver.getInstance(it) } ?: return emptyArray()
-        return resolver.findPsiElements(project, handler, GlobalSearchScope.allScope(project))
+
+        // Don't search through ".aws-sam" folders
+        val samBuildFileScopes = GlobalSearchScope.filesScope(project, findSamBuildContents(project))
+        val excludeSamBuildFileScopes = GlobalSearchScope.notScope(samBuildFileScopes)
+        val scope = GlobalSearchScope.allScope(project).intersectWith(excludeSamBuildFileScopes)
+
+        val elements = resolver.findPsiElements(project, handler, scope)
+
+        logHandlerPsiElements(handler, elements)
+
+        return elements
     }
 
     fun isHandlerValid(project: Project, runtime: Runtime, handler: String): Boolean = ReadAction.compute<Boolean, Throwable> {
         runtime.runtimeGroup?.let {
             LambdaHandlerResolver.getInstance(it)
         }?.isHandlerValid(project, handler) == true
     }
+
+    fun findSamBuildContents(project: Project): Collection<VirtualFile> =
+        ModuleManager.getInstance(project).modules.flatMap { findSamBuildContents(it) }
+
+    fun findSamBuildContents(module: Module): Collection<VirtualFile> =
+        ModuleRootManager.getInstance(module).contentRoots.map {
+            it.findChild(SamCommon.SAM_BUILD_DIR)
+        }.filterNotNull()
+            .flatMap {
+                VfsUtil.collectChildrenRecursively(it)
+            }
+
+    private fun logHandlerPsiElements(handler: String, elements: Array<NavigatablePsiElement>) = LOG.debug {
+        elements.joinToString(
+            prefix = "Found ${elements.size} PsiElements for Handler: $handler\n",
```

Comment:
nit: this will insert two line returns after the prefix, is that intended?

Reply:
The separator is only used in between items in `elements`. There is only one newline between prefix and the joined text.

Reply:
Chris is right,
```
    buffer.append(prefix)
    var count = 0
    for (element in this) {
        if (++count > 1) buffer.append(separator)
        if (limit < 0 || count <= limit) {
            buffer.appendElement(element, transform)
        } else break
    }
    if (limit >= 0 && count > limit) buffer.append(truncated)
    buffer.append(postfix)
    return buffer
```